homeassistant/components/group/sensor.py

```diff
@@ -0,0 +1,411 @@
+                None,
+            )
+        ]
+    )
+
+
+def calc_min(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate min value."""
+    val: float | None = None
+    entity_id: str | None = None
+    for sensor_id, sensor_value, _ in sensor_values:
+        if val is None or val > sensor_value:
+            entity_id, val = sensor_id, sensor_value
+
+    attributes = {ATTR_MIN_ENTITY_ID: entity_id}
+    if TYPE_CHECKING:
+        assert val is not None
+    return attributes, val
+
+
+def calc_max(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate max value."""
+    val: float | None = None
+    entity_id: str | None = None
+    for sensor_id, sensor_value, _ in sensor_values:
+        if val is None or val < sensor_value:
+            entity_id, val = sensor_id, sensor_value
+
+    attributes = {ATTR_MAX_ENTITY_ID: entity_id}
+    if TYPE_CHECKING:
+        assert val is not None
+    return attributes, val
+
+
+def calc_mean(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate mean value."""
+    result = [sensor_value for _, sensor_value, _ in sensor_values]
+
+    value: float = round(statistics.mean(result), round_digits)
+    return {}, value
+
+
+def calc_median(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate median value."""
+    result = [sensor_value for _, sensor_value, _ in sensor_values]
+
+    value: float = round(statistics.median(result), round_digits)
+    return {}, value
+
+
+def calc_last(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate last value."""
+    last_updated: datetime | None = None
+    for entity_id, state_f, state in sensor_values:
+        if last_updated is None or state.last_updated > last_updated:
+            last_updated = state.last_updated
+            last = state_f
+            last_entity_id = entity_id
+
+    attributes = {ATTR_LAST_ENTITY_ID: last_entity_id}
+    return attributes, last
+
+
+def calc_range(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate range value."""
+    result = [sensor_value for _, sensor_value, _ in sensor_values]
+
+    value: float = round(max(result) - min(result), round_digits)
+    return {}, value
+
+
+def calc_sum(
+    sensor_values: list[tuple[str, float, State]], round_digits: int
+) -> tuple[dict[str, Any], float]:
+    """Calculate a sum of values."""
+    result = 0.0
+    _LOGGER.debug("sensor_values: %s", sensor_values)
+    for _, sensor_value, _ in sensor_values:
+        result += sensor_value
+
+    value: float = round(result, round_digits)
+    return {}, value
+
+
+CALC_TYPES: dict[str, Callable] = {
```

Comment:
Can we type `Callable` more precisely?

```suggestion
CALC_TYPES: dict[
    str, Callable[[list[tuple[str, float, State]], int], tuple[dict[str, str], float]]
] = {
```